lib/urlwatch/filters.py

```diff
@@ -330,6 +331,46 @@ def filter(self, data, subfilter):
                          method=method, options=options)
 
 
+class Csv2TextFilter(FilterBase):
+    """Convert CSV to plaintext"""
+
+    __kind__ = 'csv2text'
+
+    __supported_subfilters__ = {
+            'format_message': 'A format string with the headers that will be outputted for each csv '
+                                                         'line (header will be lower-cased)',
+            'ignore_header': 'If your format string is number based, but the CSV has headers, '
+                                                       'this flag will force ignoring the header.',
+        }
+
+    __default_subfilter__ = 'format_message'
+
+    def filter(self, data, subfilter):
+        has_header = csv.Sniffer().has_header(data)
+
+        reader = csv.reader(data.split('\n'))
+        data_list = list(reader)
+        header = None
+
+        if has_header:
+            header = data_list[0]
+            data_list = data_list[1:]
+
+        header = [i.lower() for i in header]
+        message = subfilter['format_message']
```

Comment:
Should there be a default / fallback for `format_message`?

How does the CSV output look like?

Reply:
It is hard to create a default, since the main ability of this is to modify the input (csv) to a pretty string that uses rows(/headers) as format data.
If I must think about something maybe the default with headers should be:
```
First header: {first_header_data}. Second header: {second_header_data}. ...
```

Without headers the csv itself is as much "default" as we can get (it doesn't make sense to omit or reorder the columns), so no reason to use this filter.

Do you think the no format string default with header is interesting enough to add?

Reply:
Yeah, I guess there's no point in having a default then. The docs should list a good example value for this, though.